darwin/cli_functions.py

```diff
@@ -73,8 +77,8 @@ def authenticate(api_key: str, default_team: Optional[bool] = None, datasets_dir
         API key to use for the client login
     default_team: bool
```

Comment:
This PR focuses on types. The documentation will be fixed later on, once I have all the types working properly (or at least working to the extent where I can, because we know there are type inconsistencies that can only be fixed by re-writes in some areas).

It is in that second pass that I will pay more attention to documentation and enforce a standard. Right now I am mostly ignoring it.

Reply:
I have still updated it.